src/api/controllers/db-controller.ts

```diff
@@ -467,15 +467,14 @@ export async function getRosettaBlockFromDataStore(
   blockHeight?: number
 ): Promise<FoundOrNot<RosettaBlock>> {
   return await db.sqlTransaction(async sql => {
-    let query;
+    let blockQuery: FoundOrNot<DbBlock>;
     if (blockHash) {
-      query = db.getBlock({ hash: blockHash });
+      blockQuery = await db.getBlock({ hash: blockHash });
     } else if (blockHeight && blockHeight > 0) {
-      query = db.getBlock({ height: blockHeight });
+      blockQuery = await db.getBlock({ height: blockHeight });
     } else {
-      query = db.getCurrentBlock();
+      blockQuery = await db.getCurrentBlock();
     }
-    const blockQuery = await query;
 
```

Comment:
These synchronous calls to `db.get...` _before_ performing an `await` likely broke the implicit sql transaction handling where we leverage AsyncLocalStorage. IIUC, synchronous invocations of functions that return a promise break the async context.